src/sidebar/util/page-range.ts

```diff
@@ -1,36 +1,108 @@
+/**
+ * Parse a page number or range into a `[start, end]` integer pair. The `start`
+ * or `end` may be `null` if the range is open.
+ *
+ * Returns `null` if the page range could not be parsed.
+ */
+function parseRange(range: string): [number | null, number | null] | null {
+  let start;
+  let end;
+
+  if (range.includes('-')) {
+    [start, end] = range.split('-');
+  } else {
+    start = range;
+    end = range;
+  }
```

Comment:
Perhaps these can be combined:

```suggestion
  const [start, end] = range.includes('-') ? range.split('-') : [range, range];
```